java/code/src/com/redhat/rhn/frontend/action/rhnpackage/LockPackageAction.java

```diff
@@ -104,16 +104,19 @@ public ActionForward execute(ActionMapping mapping,
 
         DataResult<PackageListItem> lockedPackagesResult =
                 PackageManager.systemLockedPackages(sid, null);
-        for (PackageListItem pkg : lockedPackagesResult) {
-            if (!context.isSubmitted()) {
+
+        if (!context.isSubmitted()) {
+            for (PackageListItem pkg : lockedPackagesResult) {
                 // pre-select locked
                 pkgsToSelect.add(pkg.getIdCombo() + "~*~" + pkg.getNvrea());
             }
+        }
 
-            Package lockedPkg = PackageManager.lookupByIdAndUser(pkg.getPackageId(), user);
-            if (lockedPkg != null) {
-                pkgsAlreadyLocked.add(lockedPkg);
-            }
+        List<Package> pkgsFindAlreadyLocked = PackageManager.lookupByIdAndUser(
+                lockedPackagesResult.stream().map(PackageListItem::getPackageId)
+                        .collect(Collectors.toList()), user);
+        if (pkgsFindAlreadyLocked != null) {
```

Comment:
Null check is actually no longer needed, since ```lookupByIdAndUser``` always returns a non-null value after your last change.